web/src/stream_pill.ts

```diff
@@ -12,29 +12,41 @@ export type StreamPill = {
     stream_sub_count?: number;
 };
 
-type StreamPillWidget = InputPillContainer<StreamPill>;
+export type StreamPillWidget = InputPillContainer<StreamPill>;
 
 export type StreamPillData = StreamSubscription & {type: "stream"};
 
-function stream_display_pill(sub: StreamSubscription): string {
+export function stream_display_pill(sub: StreamSubscription): string {
     return "#" + sub.name;
 }
 
 export function create_item_from_stream_name(
     stream_name: string,
     current_items: CombinedPillItem[],
+    stream_prefix_required = true,
+    invite_streams = false,
+    show_stream_sub_count = true,
 ): InputPillItem<StreamPill> | undefined {
     stream_name = stream_name.trim();
-    if (!stream_name.startsWith("#")) {
-        return undefined;
+    if (stream_prefix_required) {
+        if (!stream_name.startsWith("#")) {
+            return undefined;
+        }
+        stream_name = stream_name.slice(1);
     }
-    stream_name = stream_name.slice(1);
 
     const sub = stream_data.get_sub(stream_name);
     if (!sub) {
         return undefined;
     }
 
+    if (invite_streams) {
```

Comment:
One approach here could be to pass a allowed streams list here. Or some validation function like `can_subscribe_others` here. This would mean that that could be used in future for cases other than invite UI.

Reply:
Yeah I passed a validation function as param.